src/components/Amalgamation/ResultingBusinessName.vue

```diff
@@ -137,14 +140,30 @@ export default class ResultingBusinessName extends Mixins(AmalgamationMixin, Nam
   onUpdateCompanyName (name: string): void {
     this.setCorrectNameOption(this.formType)
     this.setNameRequestApprovedName(name)
+
+    // if adopting a business' name, also adopt its legal type
+    if (this.formType === CorrectNameOptions.CORRECT_AML_ADOPT) {
+      const business = this.getAmalgamatingBusinesses.find(b =>
+        (b.type === AmlTypes.LEAR && b.name === name)
+      )
+      if (business?.type === AmlTypes.LEAR) {
+        this.setEntityType(business.legalType)
+      }
+    }
```

Comment:
Line 149 may seem redundant but the language service needs it to ensure that "business" is the right type of object and has a legalType property. Without this, VS Code shows an error on line 150.

I tried a few ways to organize this code and avoid apparent duplication but this is the cleanest I could make it.

Besides, line 150 also verifies that "business" was found by line 146 (ie, is not undefined).